cirq-google/cirq_google/api/v2/run_context.proto

```diff
@@ -127,15 +127,37 @@ message DeviceParametersDiff {
     // as index into the strs repeated field.
     int32 name = 2;
   }
+
+  // Param value whose type is not among proto field types supported by
+  // ArgValue. In other words, it is the responsibility of the client codes
+  // to enforce the validity and type consistency of this param value.
+  message GenericValue {
+    // Type description of the value representation. Eg. if the following value
+    // bytes field is a JSON string, type_descriptor can be its JSON namespace;
+    // or if the following value field is a protobuf serialization, type_descriptor
+    // can be its protobuf definition URL.
+    string type_descriptor = 1;
+
+    // The value in client's encoding.
+    bytes value = 2;
+  }
+
   message Param {
     // the resource group hosting this parameter key, as index into groups
     // repeated field.
     int32 resource_group = 1;
     // name of this param, as index into the strs repeated field.
     int32 name = 2;
-    // this param's new value, as message ArgValue to allow variant types,
-    // including bool, string, double, float and arrays.
-    ArgValue value = 3;
```

Comment:
Is changing an existing field to a oneof backwards compatible?
This is an actively used proto, so we do need to be careful about backwards compatibility.

Reply:
this is completely ok. the old wire format is compatible with the oneof wire format, as long as we do not change field id of the field (ArgValue value) which is moved to a oneof.
pls see the accepted answer of https://yaqs.corp.google.com/eng/q/4770647064182784